### PR TITLE
Modified Fast R-CNN scripts for distributed learning.

### PR DESCRIPTION
**Pull request by kyoro1**
In accordance with discussions [here](https://github.com/Microsoft/CNTK/issues/2354#issuecomment-331634165) and [here](https://github.com/Microsoft/CNTK/issues/2354#issuecomment-332055051), I generated "distributed" Fast R-CNN, and changed `PARAMETERS.py` and `README.md` accordingly.

### CODE REVIEW

File: Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py [hunk 1]
```diff
@@ -56,6 +56,10 @@ def __init__(self, datasetName):
         self.cntk_max_epochs = -1                # set per data set below
         self.cntk_momentum_time_constant = -1    # set per data set below
 
+        # for Distributed learner
+        self.num_quantization_bits = 1   # set for distributed learner
```
Comment (KeDengMS) [inline Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py]: I'd start from 32-bit data parallel SGD for accuracy, unless communication cost is higher than computation.

File: Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py [hunk 2]
```diff
@@ -56,6 +56,10 @@ def __init__(self, datasetName):
         self.cntk_max_epochs = -1                # set per data set below
         self.cntk_momentum_time_constant = -1    # set per data set below
 
+        # for Distributed learner
+        self.num_quantization_bits = 1   # set for distributed learner 
+        self.warm_up = 1                 # set for distributed learner
```
Comment (KeDengMS) [inline Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py]: warm_up = 1 seems too small for 1-bit SGD. For 32-bit data-parallel SGD you can set it to zero though.

File: Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
+﻿# Copyright (c) Microsoft. All rights reserved.
+
+# Licensed under the MIT license. See LICENSE.md file in the project root
+# for full license information.
+# ==============================================================================
+
+from __future__ import print_function
```
Comment (KeDengMS) [inline Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py]: Is this file copied from non-distributed version? Code copy would increase maintenance cost in future, please either modify the non-distributed version to support parallel training, or import functions from it to reduce code duplication.

File: Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    progress_printer = ProgressPrinter(tag='Training', num_epochs=max_epochs)
+    trainer = Trainer(frcn_output, (ce, pe), learner, progress_printer)
+
+    # Get minibatches of images and perform model training
+    print("Training Fast R-CNN model for %s epochs." % max_epochs)
+    log_number_of_parameters(frcn_output)
+    for epoch in range(max_epochs):       # loop over epochs
+        sample_count = 0
+        while sample_count < epoch_size:  # loop over minibatches in the epoch
+            data = minibatch_source.next_minibatch(min(mb_size, epoch_size-sample_count), input_map=input_map)
```
Comment (KeDengMS) [inline Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py]: This would feed the same data to all workers, which effectively just scales up learning rate. Please use partitions.

File: Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py (hunk 3 above)
Comment (kyoro1) [reply Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py]: As you see in [this script](https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Classification/ConvNet/Python), there are similar scripts for usual learning and distributed one. As a policy of CNTK repository, should we merge this script to one?

File: Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py (hunk 2 above)
Comment (kyoro1) [reply Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py]: Are there any reference as to distributed learning? I didn't find the detailed specification on it.

File: Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py (hunk 4 above)
Comment (kyoro1) [reply Examples/Image/Detection/FastRCNN/BrainScript/A2_RunWithPyModel_distributed.py]: Are there any reference as to distributed learning? I didn't find the detailed specification on it.

File: Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py (hunk 1 above)
Comment (kyoro1) [reply Examples/Image/Detection/FastRCNN/BrainScript/PARAMETERS.py]: Are there any reference as to distributed learning? I didn't find the detailed specification on it.

**Comment by kyoro1** [comment]
@KeDengMS Updated [newer one](https://github.com/Microsoft/CNTK/pull/2414/commits/0b55cba064517985a4a99d728caf4baa230d3541). Can you check it now? Just in case, it includes only changes for learner, because the method 'training_session' is under verification.

**Comment by kyoro1** [comment]
Related PR https://github.com/Microsoft/CNTK/pull/2561  was closed.
